mne_sandbox/preprocessing/_sns.py

```diff
@@ -112,7 +115,11 @@ def fit(self, raw, verbose=None):
             # XXX Eventually we might want to actually threshold here (with
             # rank-deficient data it could matter)
             eigval, eigvec = _pca(data_cov[np.ix_(idx, idx)], thresh=None)
-            eigvec *= 1. / np.sqrt(eigval)
+            # Some of the eigenvalues could be zero, don't let it blow up
+            norm = np.zeros(len(eigval))
+            use_mask = eigval > 0
+            norm[use_mask] = 1. / np.sqrt(eigval[use_mask])
+            eigvec *= norm
```

Comment:
looks reasonable. I suppose it's the equivalent of L38 [here](http://audition.ens.fr/adc/NoiseTools/src/NoiseTools/doc/NoiseTools/nt_sns0.html) ?